File: Extension Objects/Runtime Ports/_Source Codes (XNA & Java)/XNA/CRuncommentor.cs
// commentor object - XNA port (All Platforms)
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using RuntimeXNA.Extensions;
using RuntimeXNA.Services;
using RuntimeXNA.RunLoop;
using RuntimeXNA.Sprites;
using RuntimeXNA.Conditions;
using RuntimeXNA.Actions;
using RuntimeXNA.Expressions;
using RuntimeXNA.Objects;
using RuntimeXNA.Params;
using RuntimeXNA.Frame;
using RuntimeXNA.OI;
using RuntimeXNA.Movements;

namespace RuntimeXNA.Extensions
{
    class CRuncommentor : CRunExtension
    {
        const int CND_COMMENT = 0; // Huh, no CND_LAST
        
        const int ACT_COMMENT = 0;

        public override int getNumberOfConditions()
        {
            return 1;
        }

        public override bool createRunObject(CFile file, CCreateObjectInfo cob, int version)
        {
            return true;
        }

        public override bool condition(int num, CCndExtension cnd)
        {
            switch (num)
            {
                case CND_COMMENT:
                    return true;
            }
            return false;//won't happen
        }

        public override void action(int num, CActExtension act)
        {
            switch (num)
            {
                case ACT_COMMENT:
                    break;
            }
        }

        public override CValue expression(int num)
        {
            switch (num)
            {
                // no expressions
            }
            return new CValue(0);// (Usually) won't be used
        }
    }
}

File: Extension Objects/Runtime Ports/_Source Codes (XNA & Java)/Java/CRuncommentor.java
import java.awt.*;
import java.io.*;
import Extensions.*;
import Services.*;
import RunLoop.*;
import Conditions.*;
import Actions.*;
import Expressions.*;
import java.util.ArrayList;

public class CRuncommentor extends CRunExtension
{
    // <editor-fold defaultstate="collapsed" desc=" A/C/E Constants ">
    public static final int CNDCOMMENT = 0;
    public static final int CNDLAST = 1;

    public static final int ACTCOMMENT = 0;

    // </editor-fold>

    public CRuncommentor()
    {

    }
    public @Override int getNumberOfConditions()
    {
	    return CNDLAST;
    }
    public @Override boolean createRunObject(CBinaryFile file, CCreateObjectInfo cob, int version)
    {
        return false;
    }
    public @Override void destroyRunObject(boolean bFast)
    {

    }


    public @Override int handleRunObject()
    {
        return 0;
    }


    public @Override void displayRunObject(Graphics2D g2)
    {
    }
    public @Override boolean saveRunObject(DataOutputStream stream)
    {
        return true;
    }
    public @Override boolean loadRunObject(DataInputStream stream)
    {
        return true;
    }

    // Conditions
    // -------------------------------------------------
    public @Override boolean condition(int num, CCndExtension cnd)
    {
        switch (num)
        {
            case CNDCOMMENT:
                return true;
        }
        return false;
    }

    // Actions
    // -------------------------------------------------
    public @Override void action(int num, CActExtension act)
    {
        switch (num)
        {
            case ACTCOMMENT:
                break;
        }
    }

    // Expressions
    // -------------------------------------------------
    public @Override CValue expression(int num)
    {
        switch (num)
        {

        }
        return null;
    }
}